infer/lib/python/inferlib/capture/gradle.py

```diff
@@ -31,9 +29,92 @@ def gen_instance(*args):
 create_argparser = util.base_argparser(MODULE_DESCRIPTION, MODULE_NAME)
 
 
+def extract_filepath(parts):
+    if not parts:
+        return ([], None)
+    path = ' '.join(parts)
+    if os.path.isfile(path):
+        return ([], path)
+    remainder, path = extract_filepath(parts[1:])
+    return ([parts[0]] + remainder, path)
+
+
+# Please run the doctests using:
+# $ python -m doctest -v gradle.py
+def extract_all(javac_arguments):
+    """Extract Java filenames and Javac options from the Javac arguments.
+
+    >>> os.path.isfile = lambda s: s[1:].startswith('path/to/')
+    >>> extract_all([])
+    {'files': [], 'opts': []}
+    >>> extract_all(['-opt1', 'optval1', '/path/to/1.java'])
+    {'files': ['/path/to/1.java'], 'opts': ['-opt1', 'optval1']}
+    >>> extract_all(['-opt1', 'optval1', '/path/to/a', 'b/1.java'])
+    {'files': ['/path/to/a b/1.java'], 'opts': ['-opt1', 'optval1']}
+    >>> extract_all(['-opt1', 'opt', 'val1', '/path/to/1.java'])
+    {'files': ['/path/to/1.java'], 'opts': ['-opt1', 'opt val1']}
+    >>> extract_all(['-opt1', '/path/to/a', 'b/c', 'd/1.java', '-opt2'])
+    {'files': ['/path/to/a b/c d/1.java'], 'opts': ['-opt1', '-opt2']}
+    >>> extract_all(['-opt1', 'optval1', '-path/to/1.java'])
+    {'files': ['-path/to/1.java'], 'opts': ['-opt1', 'optval1']}
+    >>> extract_all(['-opt1', 'optval1', '/path/to/', '-1.java'])
+    {'files': ['/path/to/ -1.java'], 'opts': ['-opt1', 'optval1']}
+    >>> extract_all(['undef1', 'undef2'])
+    {'files': [], 'opts': ['undef1 undef2']}
```

Comment:
Frankly, I'm not sure what to do here. By the javac docs, besides options and java files we may have classes and @argfiles, for the former it's better to split, for the latter - join. I think it's better to improve the logic, and assume that if we don't have any @argfiles, the leftovers are the names of the classes. It will be addressed in the up-to-dated PR.